### PR TITLE
use pytest-recording with VCR

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Background

use pytest-recording
### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by merwanehamadi** [comment]
@Pwuts @collijk

File: tests/conftest.py [hunk 1]
```diff
@@ -36,3 +37,16 @@ def api_manager() -> ApiManager:
     api_manager_.reset()
     yield api_manager_
     api_manager_.__dict__.update(old_attrs)
+
+
+@pytest.fixture
+def vcr_config():
```
Comment (collijk) [inline tests/conftest.py]: Can you write a comment about how this works for those of us not familiar.  I typically expect fixtures to be invoked some where or to have `@pytest.fixture(autouse=True)`

**Comment by collijk** [comment]
I would put your `vcr_config` feature and the functions it calls in `tests/integration/conftest.py`.  pytest sources those things hierarchically, and I think we want to limit testing utilities and the like to all live in `conftest.py` files if we can.

**Review by collijk — CHANGES_REQUESTED** [review]
Two small things, but this otherwise looks quite good.

**Pull request merged**